rsmtool/rsmpredict.py

```diff
@@ -25,6 +28,151 @@
+
+    The main difference between this function and the ``compute_and_save_predictions()``
+    function is that the former is meant for batch prediction and reads
+    all its inputs from disk and writes its outputs to disk. This function,
+    however, is meant for real-time inference rather than batch. To this end,
+    it operates *entirely* in memory. Note that there is still a bit of overlap
+    between the two computation paths since we want to use the RSMTool API
+    as much as possible.
+
+    This function should only be used when the goal is to generate predictions
+    using RSMTool models in production. The user should everything from disk
+    in a separate thread/function and pass the inputs to this function.
+
+    Parameters
+    ----------
+    input_features : dict[str, float]
+        A dictionary containing the features for the instance for which to
+        generate the model predictions. The keys should be names of the
+        features on which the model was trained and the values should
+        be the *raw* feature values.
+    modeler : rsmtool.modeler.Modeler object
+        The RSMTool modeler object using which the predictions
+        are to be generated. This object should be created from the already
+        existing model file in the "output" directory of the previously run
+        RSMTool experiment.
+    df_feature_info : pandas DataFrame
+        A pandas dataframe containing the information regarding the model features.
+        The index of the dataframe should be the names of the features and
+        the columns should be:
+
+        - "sign" : 1 or -1.  Indicates whether the feature value needs to
+          be multiplied by -1.
+        - "transform" : :ref:`transformation <select_transformations_rsmtool>`
+          that needs to be applied to this feature.
+        - "train_mean", "train_sd" : mean and standard deviation for outlier
+          truncation.
+        - "train_transformed_mean", "train_transformed_sd" : mean and standard
+          deviation for computing z-scores.
+
+        This dataframe should be read from the "feature.csv" file under the
+        "output" directory of the previously run RSMTool experiment.
+    df_postprocessing_params : pandas DataFrame
+        A pandas dataframe containing the parameters needed for
+        post-processing the raw prediction. This dataframe should be read from the
+        "postprocessing_params.csv" file under the "output" directory of the
+        previously run RSMTool experiment.
+    predict_expected : bool, optional
+        Predict expected scores for classifiers that return probability
+        distributions over score. Note also that this assumes that the score
+        range consists of contiguous integers. Defaults to ``False``.
+    logger : logging object, optional
+        A logging object. If ``None`` is passed, get logger from ``__name__``.
+        Defaults to ``None``.
+
+    Returns
+    -------
+    pandas DataFrame
```

Comment:
It's not a very big deal, but could we just return a dictionary since it's really just one `Series` we're getting?